exonum/src/node/consensus.rs

```diff
@@ -660,9 +707,14 @@ impl NodeHandler {
 
         let full_proposes = self.state.check_incomplete_proposes(hash);
         // Go to handle full propose if we get last transaction.
+        let mut height_bumped = false;
         for (hash, round) in full_proposes {
             self.remove_request(&RequestData::ProposeTransactions(hash));
-            self.handle_full_propose(hash, round);
+            // If a new height was achieved, no more proposals for this height
+            // should be processed. However, we still have to remove requests.
```

Comment:
Can the logic further down the method (i.e., handling a full block) work incorrectly if the new height is achieved here?

Reply:
Don't think so: if we're involved in the processing of the new height approval, we shouldn't be lagging behind, and thus shouldn't request blocks.

Reply:
I added a comment explaining this, hopefully helpful. However, this is a place for a future refactoring.